routers/web/repo/view.go

```diff
@@ -342,7 +331,7 @@ func renderReadmeFile(ctx *context.Context, readmeFile *namedBlob, readmeTreelin
 		ctx.Data["EscapeStatus"], ctx.Data["FileContent"], err = markupRender(ctx, &markup.RenderContext{
 			Ctx:          ctx,
 			RelativePath: path.Join(ctx.Repo.TreePath, readmeFile.name), // ctx.Repo.TreePath is the directory not the Readme so we must append the Readme filename (and path).
-			URLPrefix:    readmeTreelink,
+			URLPrefix:    path.Dir(treeLink),
```

Comment:
I need some advice on this. `readmeTreelink` was an annoying appendix that needed to be carried around like an albatross on our shoulders. I believe the only reason it existed  in the `docs/README.md` case, where you need to append "docs/" to treeLink to maintain the invariant that URLPrefix is the full directory path (including '/src/' and the branch name) for the file being rendered.

But as far as I can tell, in the README case, the contents of `URLPrefix` are never served to the user, and nothing seems broken with it switched back to the default. READMEs are a special case -- rendered inline attached to a tree, never alone (unless the user has specifically opened the README file, in which case findReadmeFile() and renderReadmeFile() aren't used). I _think_ this is okay. But maybe I've missed something and it is important; it could be recomputed based on `path.Dir(readmeFile.name)`, which carries the subdirectory if one exists, just for safety. Any advice?